test/Constraints/generics.swift

```diff
@@ -528,3 +528,11 @@ do {
     _ = { baz($0) }(construct_generic { d }) // Ok
   }
 }
+
+// rdar://problem/35541153 - Generic parameter inference bug
+
+func rdar35541153() {
+  func foo<U: Equatable, V: Equatable, C: Collection>(_ c: C) where C.Element == (U, V) {}
```

Comment:
Could you add a second example here where the the function in question takes two parameters? I want to be sure that `ParenType` and `TupleType` function inputs don't end up getting different behavior.

Reply:
Sure!